RoboJackets/robocup-software
# Makefile tweak

## Description
**Ignore everything in docs/build/. These files were autogenerated and do not need to be reviewed.**

Adds helpful shortcuts to our root Makefile for a better development experience. (Note: many binds Ubuntu-only, like our repo as a whole.) I would appreciate comments on both the function and the readability of the docs describing them.

Summary list:
 * `make run-*` - runs the specified version of code, e.g. `make run-sim` or `make run-soccer` (sim2play is still broken on this branch)
 * `make docs` - builds docs statically (local copy, not hosted on the internet)
 * `make open-docs` - opens statically built docs
 * `source ./source.bash` or `. ./source.bash` - sources ros2 and install in one command
 * `make again` - if repo built already with another method, builds faster by skipping CMake files

See Steps to Test for more detail.

## Steps to test
### Test Case 1
1. Run `make docs`.
2. Run `make open-docs`.
3. Click on the "Getting Started" link of the webpage that opens up, and read through and try to use the new binds.

Expected result: Everything works as described in the documentation.
